src/main/java/io/reactivex/internal/operators/observable/ObservableReplay.java

```diff
@@ -617,23 +617,20 @@ final void removeFirst() {
             // can't just move the head because it would retain the very first value
             // can't null out the head's value because of late replayers would see null
             setFirst(next);
-        }
```

Comment:
When testing you can just call `removeFirst()` repeatedly. No need to introduce a test-only method. Plus, doing this reduces some memory leak false positives.